src/EventGridExtension/EventGridExtensionConfig.cs

```diff
@@ -43,6 +44,12 @@ public void Initialize(ExtensionConfigContext context)
                 .AddOpenConverter<JObject, OpenType.Poco>(typeof(JObjectToPocoConverter<>))
                 .BindToTrigger<JObject>(new EventGridTriggerAttributeBindingProvider(this));
 
+            // Register the output binding
+            context
+                .AddBindingRule<EventGridAttribute>()
+                .WhenIsNotNull(nameof(EventGridAttribute.SasKey))
+                .WhenIsNotNull(nameof(EventGridAttribute.TopicEndpointUri))
```

Comment:
`WhenIsNotNull()` is a [filter](https://github.com/Azure/azure-webjobs-sdk/blob/08709277db41b8dac7cb1e5835005a800af4f304/src/Microsoft.Azure.WebJobs.Host/Bindings/BindingProviders/FilterNode.cs), they are used to distinguish between [different rules](https://github.com/Azure/azure-webjobs-sdk-extensions/blob/b68ec64d7a3fa38c0735ca167e77fcd583cd3311/src/WebJobs.Extensions.CosmosDB/Config/CosmosDBConfiguration.cs#L69-L79) applied to the same attribute (in this case, if saskey is missing, we won't throw an exception, instead, we will keep looking for suitable rules). What we want is probably a validation...I am still looking at proper ways, some candidates are [AddValidator](https://github.com/Azure/azure-webjobs-sdk-extensions/blob/b68ec64d7a3fa38c0735ca167e77fcd583cd3311/src/WebJobs.Extensions.CosmosDB/Config/CosmosDBConfiguration.cs#L64), [ValidationAttribute](https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs/BlobNameValidationAttribute.cs)

Reply:
ah thanks for this, yes what i was thinking was that it should error if those aren't present on the attribute; will look in to the validator/validation approach.